pandas/core/ops/__init__.py

```diff
@@ -888,9 +881,11 @@ def wrapper(self, other, axis=None):
                 )
 
         result = self._constructor(res_values, index=self.index)
-        # rename is needed in case res_name is None and result.name
-        #  is not.
-        return finalizer(result).rename(res_name)
+        result = finalizer(result)
+
+        # pin name for case res_name is None and result.name is not.
```

Comment:
this comment is still not correct I think.  As far as I know, result.name will always be None? 
As mentioned in the original PR (https://github.com/pandas-dev/pandas/pull/27803/files#r311961558), I think it is here about self.name not being None.

The problem we were trying to solve is that finalize will put back the original name, which might be wrong. So that's the reason we can't pass `res_name` into `_constructor` and afterwards do `finalize`, as that might change the name again.